### PR TITLE
Add initial documentation for --vm-driver=none

### LINKED ISSUE #2575: More documentation around vm-driver=none for local use
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): FEATURE REQUEST

Please provide the following details:

**Environment**:

**Minikube version** (use `minikube version`): v0.25.0
- **OS** (e.g. from /etc/os-release): "Ubuntu 16.04.3 LTS (Xenial Xerus)"
- **VM Driver** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep DriverName`): none
- **ISO version** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep -i ISO` or `minikube ssh cat /etc/VERSION`): n/a
- **Install tools**:
- **Others**: Docker version 18.02.0-ce

**What happened**: I spent a long time struggling to get things working properly with vm-driver=none

**What you expected to happen**: Find documentation which provides coverage for this very valuable use-case.

**Anything else do we need to know**:

I have a linux laptop that I want to use k8s on. I do not want to run a VM, since that is an inefficient use of my laptop's memory (always either too much or too little memory allocated for the given docker containers I want to run) and slower due to the extra virtualization. Also it shouldn't be necessary - I have a Linux environment, why should I have to run another VM just to run Linux? That's why I bought this laptop.

The use of vm-driver=none is hardly documented at all. What little there is does not seem to consider the value for developer machines.

Yesterday I filed https://github.com/kubernetes/minikube/issues/2571, but this was actually not the full story. I had a red-herring with IP addresses which led me to believe that the docker0 interface was the root of the problem. It turns out, the real problem was whatever IP address I happened to have bound to my ethernet interface would be used in the construction of the cluster. If that IP address changed for any reason (as often happens on laptops) then the whole environment was inaccessible.

The workaround was not to specify a bridge IP for docker, as I had thought. Instead you need to start minikube like so:

    minikube start --vm-driver=none --apiserver-ips 127.0.0.1 --apiserver-name localhost

And then go and edit ~/.kube/config, replacing the server IP that was detected from the main network interface with "localhost". For example, mine now looks like this:

```
- cluster:
    certificate-authority: /home/jfeasel/.minikube/ca.crt
    server: https://localhost:8443
  name: minikube
```

With this configuration, I can access my local cluster all of the time, even if the main network interface is disabled.

Also, we should note that it is required to have "socat" installed on the Linux environment. See this issue for details: https://github.com/kubernetes/kubernetes/issues/19765  I saw this when I tried to use helm to connect to my local cluster; I got errors with port-forwarding. Since I'm using Ubuntu all I had to do was `sudo apt-get install socat` and then everything worked as expected.

### PR DESCRIPTION
**Pull request by tstromberg**
Resolves #2575

### CODE REVIEW

**Comment by bhack** [comment]
What it is the status of https://github.com/kubernetes/minikube/pull/2926?
Cause probably we can suggest a lighter solution for users that are looking for none.

File: docs/vmdriver-none.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+Running `minikube --vm-driver=none` outside of a VM could result in data loss, system instability and decreased security.
+
+minikube was designed to run within a dedicated VM, and executes arbitrary processes as root, and when used with `--vm-driver=none`, may overwrites system binaries, configuration files, and system logs. 
+
+Usage of `--vm-driver=none` outside of a VM could also result in services being exposed in a way that may make them accessible to the public internet. Even if your host is protected by a firewall, these services still be vulnerable to [CSRF](https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)) or [DNS rebinding](https://en.wikipedia.org/wiki/DNS_rebinding) attacks.
+
+## Known Issues
+
+* You cannot run more than one `--vm-driver=one` instance within a VM #2781
```
Comment (jakefeasel) [inline docs/vmdriver-none.md]: typo - should be `--vm-driver=none`

**Comment by jakefeasel** [comment]
Sad as this documentation makes me, at least it does clarify the intent for this feature. Thanks.

**Comment by tstromberg** [comment]
> Sad as this documentation makes me, at least it does clarify the intent for this feature. Thanks.
> 
> On another note, it would be really nice if we could point people who are interested in running k8s directly on their servers to other projects which would help them accomplish that goal.

You've raised a good point about a valid use case. I've updated the docs in a way which hopefully captures the conditions in which you may want to run --vm-driver outside of a VM.

**Comment by bhack** [comment]
Probably there could be also some other alternatives to the insecure none like https://github.com/kubernetes/minikube/issues/3244

**Comment by RA489** [comment]
@tstromberg please squash the commit to one.

File: docs/vmdriver-none.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+The host running `minikube --vm-driver=none` should be:
+
+* Isolated from the rest of the network with a firewall
+* Disposable and easily reprovisioned, as this mode may overwrite system binaries, configuration files, and system logs
+
+If you find yourself running a web browser on the same host running `--vm-driver=none`, please see __Should vm-driver=none be used on a personal development machine? No.__
+
+## Known Issues
+
+* You cannot run more than one `--vm-driver=none` instance on a single host #2781
```
[the remark this answers is not in the document]
Comment (tstromberg) [reply docs/vmdriver-none.md]: GitHub autolinks #digit notation to issues: https://help.github.com/articles/autolinked-references-and-urls/#issues-and-pull-requests

**Comment by tstromberg** [comment]
> @tstromberg please squash the commit to one.

Done, though I don't feel it was necessary either.

File: docs/vmdriver-none.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## Overview
+
+This document is written for system integrators who are familiar with minikube, and wish to run it within a customized VM environment.
+
+`--vm-driver=none` allows advanced minikube users to skip VM creation, allowing minikube to be run on an user-supplied VM.
+
+## What operating systems are supported?
+
+`--vm-driver=none` supports releases of Debian, Fedora, and buildroot that are less than 2 years old.
```
[the remark this answers is not in the document]
Comment (tstromberg) [reply docs/vmdriver-none.md]: I wanted to place an an upper boundary on version skew, as a defense against users coming to us with ancient releases of Fedora and expecting us to support it. 

I'm happy to get more specific later, but figure this is an improvement over nothing.

**Pull request merged**